### ISSUE TITLE
Attention OCR does not converge.

### ISSUE THREAD

**Issue by Guneetkaur03** [tensorflow/models#8310]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 19.04
- Mobile device (e.g., Pixel 4, Samsung Galaxy 10) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version (use command below):
- Python version: python 3.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Please provide the entire URL of the model you are using?**

**Describe the current behavior**

**Describe the expected behavior**

**Code to reproduce the issue**

**Other info / logs**

When I train attention OCR on fsns data, my loss is not going below 30. How does one use fine-tuning and hyperparameter optimization to reduce training loss?

Also, when the same model is trained on my custom num plate images, nearly 8k images, my loss increases and fails to converge. Any inputs on how to overcome this challenge.

**Comment by pranaysharma** [comment]
HI I too am facing issues in training the Attention OCR as my loss on the data provided does not reduce much. Any pointers will be great!!

**Comment by huynhldvn** [comment]
@Guneetkaur03 , @pranaysharma I got it too, after 34 hours, 200k step, the loss consistency 34.7327. I have ~500 images for trainning.

**Comment by R1234A** [comment]
I am also facing the same issue, the model is not converging and the loss is high. 
Any comments or suggestions regarding the same will be appreciated.

**Comment by R1234A** [comment]
@Guneetkaur03 @pranaysharma @huynhldvn I have trained the model on the custom dataset and now trying to test it. But the demo_inference.py file is kind of weird and not giving accurate results. How can I fix the issue or run my test images through the model??

Any help will be appreciated...Thanks in advance...

**Comment by xavigibert** [comment]
Regarding the first issue, the fully-converged model achieves a loss of a little bit above 30, so this is working as intended. Note that by default, the reported loss is the total loss. If you want to see the average loss across all characters in the sequence, you can pass the `--average_across_timesteps` flag to `train.py`. In that case, the loss at converge should be around 0.9.

If you are unsure about whether your training has converged, you can always run eval to check the accuracy of the model.

`python eval.py --train_log_dir=/tmp/attention_ocr/train/ --split_name=train --number_of_steps=1`
eval/CharacterAccuracy[0.988102436]
eval/SequenceAccuracy[0.934375]

`python eval.py --train_log_dir=/tmp/attention_ocr/train/ --split_name=test --number_of_steps=1`
eval/CharacterAccuracy[0.969825447]
eval/SequenceAccuracy[0.842187524]

`python eval.py --train_log_dir=/tmp/attention_ocr/train/ --split_name=validation --number_of_steps=1 --num_batches=505`
eval/CharacterAccuracy[0.973771393]
eval/SequenceAccuracy[0.8477723]

Regarding the issue about the model not converging on your custom dataset, please go to [Stack Overflow](https://stackoverflow.com/questions/tagged/tensorflow-model-garden) for help and support.

**Comment by singhSaurabh102** [comment]
@xavigibert @Guneetkaur03 @pranaysharma @huynhldvn How to makeup FSNS dataset with my own image for attention OCR tensorflow model and how to annotate data(annotation tool Name).
